### PR TITLE
Feature/l1 l2

### PR DESCRIPTION
**Pull request by Daryl-L**
https://github.com/Magickbase/ckb-explorer-public-issues/issues/656

### CODE REVIEW

File: src/components/Btc/Transaction/styles.module.scss [hunk 1]
```diff
@@ -26,6 +26,22 @@
...
+    .confirmation {
+      color: rgb(51 51 51 / 100%);
+      display: flex;
+      border-radius: 4px;
+      font-size: 12px;
+      line-height: 14px;
+      align-items: center;
+      justify-content: center;
+
+      // width: 45px;
```
Comment (Keith-CY) [inline src/components/Btc/Transaction/styles.module.scss:38]:
```suggestion
```

File: src/locales/zh.json [hunk 2]
```diff
@@ -443,7 +443,11 @@
         "btc_txid": "BTC 交易",
         "cell": "Cell",
         "cells": "Cells",
-        "view_on_bitcoin_explorer": "在 Bitcoin 浏览器查看"
+        "view_on_bitcoin_explorer": "在 Bitcoin 浏览器查看",
+        "step": {
+          "isomorphic": "Isomorphic",
+          "unlock": "Unlock"
```
Comment (Keith-CY) [inline src/locales/zh.json:449]: Should these 2 fields be translated?

**Review by Keith-CY — APPROVED** [review]

File: src/locales/zh.json (hunk 2 above)
Comment (Daryl-L) [reply src/locales/zh.json:449]: @Sven-TBD Should be translated here?

**Comment by Daryl-L** [comment]
> Visit Preview

Internal Server Error from the API service.

**Comment by FrederLu** [comment]
https://ckb-explorer-frontend-in-magickbase-repo-1ten3dc96-magickbase.vercel.app/en/transaction/0x57872f2f13d1b0e583d66d5c2ee5b9ce10d775427eddcc988d413e6a873ec72f
image
image
image

image

Please follow the prd document for RGB++ attribute text. At the same time, the `L1 ->L2` icon is missing next to `RGB++ Digest`.

**Comment by Daryl-L** [comment]
image
What should be displayed here when the transaction is neither isomorphic transaction or unlock transaction?

**Comment by Daryl-L** [comment]
> /transaction/0x57872f2f13d1b0e583d66d5c2ee5b9ce10d775427eddcc988d413e6a873ec72f

The reason of problem 1 is null returned from API

**Comment by FrederLu** [comment]
> > /transaction/0x57872f2f13d1b0e583d66d5c2ee5b9ce10d775427eddcc988d413e6a873ec72f
> 
> The reason of problem 1 is null returned from API

@Sven-TBD   Need to confirm here.

**Comment by Daryl-L** [comment]
> image
> > 1、Please follow the prd document for RGB++ attribute text. At the same time, the L1 ->L2 icon is missing next to RGB++ Digest.
> 
> ` the L1 ->L2 icon is missing next to RGB++ Digest` This problem still exists.
> 
> > 3、The RGB++ attribute on the xudt list/Address shows unusual data.
> > 4、Bitcoin TX: data is displayed repeatedly.
> > 5、Data entered as btc time lock is missing attributes and L1->L2 icon.
> 
> These issues remain to be addressed.

3. Data from API is null, how to display? 
4. Should we remove the confirmation blocks and block time from the right side? @Sven-TBD 
5. According to the data from API, it is not an RGB++ transaction @Sven-TBD please check if it isn't a RGB++ transaction, how to display?

**Comment by FrederLu** [comment]
> 3.Data from API is null, how to display?

image

Through the requirement on prd, the page does not display the label when null is returned.   cc @Sven-TBD

**Comment by Daryl-L** [comment]
> > 3.Data from API is null, how to display?
> 
> image
> Through the requirement on prd, the page does not display the label when null is returned. cc @Sven-TBD

The first condition shows if it was not a RGB++ transaction, but actually, it is.

**Comment by Daryl-L** [comment]
> > image What should be displayed here when the transaction is neither isomorphic transaction or unlock transaction? @Sven-TBD
> 
> Normally, this wont happen when this is a RGB++ Transaction. However, we can still handle this case by display nothing.

I just missed this comment, do you mean that we should not display the whole tag when it is null, even though it is a RGB++ transaction?

**Comment by Daryl-L** [comment]
> /transaction/0x57872f2f13d1b0e583d66d5c2ee5b9ce10d775427eddcc988d413e6a873ec72f

What is the tip message when hover it?

**Review by Keith-CY — APPROVED** [review]

**Comment by FrederLu** [comment]
https://ckb-explorer-frontend-in-magickbase-repo-73yzx2c9e-magickbase.vercel.app/address/ckt1qyq2c3mvt5m9rqpq4yw6y59w7h20g0vp8casg94j29
image

https://ckb-explorer-frontend-in-magickbase-repo-73yzx2c9e-magickbase.vercel.app/transaction/0x3068a4b4a57941ac41f120de3d831c8402f2a83800eebfe6dcba174e4850e1d0
image

This is an `unlock` transaction, but the attributes and tags are missing.

**Comment by Daryl-L** [comment]
> https://ckb-explorer-frontend-in-magickbase-repo-73yzx2c9e-magickbase.vercel.app/address/ckt1qyq2c3mvt5m9rqpq4yw6y59w7h20g0vp8casg94j29 image
> 
> https://ckb-explorer-frontend-in-magickbase-repo-73yzx2c9e-magickbase.vercel.app/transaction/0x3068a4b4a57941ac41f120de3d831c8402f2a83800eebfe6dcba174e4850e1d0 image
> 
> This is an `unlock` transaction, but the attributes and tags are missing.

Because it is not a RGB++ transaction.

**Comment by Daryl-L** [comment]
> @Daryl-L Can we still show these attributes even though it is not a typical RGB++ TX but the attributes and tags are missing.

It's okay, but it is not suitable for your flow chart in PRD.

**Comment by FrederLu** [comment]
https://github.com/Magickbase/ckb-explorer-frontend/assets/25676576/3a88b546-53ed-47ac-a841-c7d8e508ff2f

https://ckb-explorer-frontend-in-magickbase-repo-okomdwwgi-magickbase.vercel.app/transaction/0xd36d39eb0144a0135b21a2fdd5227b99bc6d2fc0d9d707c215e4e3fe1d585a52
image
image

The label on the side of `RGB++ Digest` lacks the `Hover` prompt information. At the same time, there are now 2 labels, and the lower ones need to be hidden.
Labels are also required next to `RGB++ Digest` on the list page to avoid duplication.

**Comment by Daryl-L** [comment]
> 2024-06-13.12.06.29.mov 
> https://ckb-explorer-frontend-in-magickbase-repo-okomdwwgi-magickbase.vercel.app/transaction/0xd36d39eb0144a0135b21a2fdd5227b99bc6d2fc0d9d707c215e4e3fe1d585a52 image image
> 
> The label on the side of `RGB++ Digest` lacks the `Hover` prompt information. At the same time, there are now 2 labels, and the lower ones need to be hidden. Labels are also required next to `RGB++ Digest` on the list page to avoid duplication.

I didn't get it from the draft, maybe it is not updated yet. Could you please make some confirmation? @Sven-TBD

**Comment by Kirl70** [comment]
> The label on the side of `RGB++ Digest` lacks the `Hover` prompt information. At the same time, there are now 2 labels, and the lower ones need to be hidden. Labels are also required next to `RGB++ Digest` on the list page to avoid duplication.

The design draft has been updated, and the labels are now aligned next to the `RGB++ Digest`.
https://www.figma.com/design/6XNoimRDbFTTNm016rbIdU/Magickbase?node-id=35067-23136&t=lGCoipesjyYjIWSO-1
image

**Comment by FrederLu** [comment]
>5、Data entered as btc time lock is missing attributes and L1->L2 icon.

The problem still exists. Data involving `BTC Time Lock` needs to be processed https://github.com/Magickbase/ckb-explorer-public-issues/issues/698 first.

---
https://ckb-explorer-frontend-in-magickbase-repo-okomdwwgi-magickbase.vercel.app/transaction/0x9099747a27e3e4f5064c211a0192323b2268d26044a81369fff51df9e77666f3
image

"leap_direction": The data for "leapoutBTC" is missing RGB++ tags.

**Comment by Daryl-L** [comment]
> > 5、Data entered as btc time lock is missing attributes and L1->L2 icon.
> 
> The problem still exists. Data involving `BTC Time Lock` needs to be processed [Magickbase/ckb-explorer-public-issues#698](https://github.com/Magickbase/ckb-explorer-public-issues/issues/698) first.
> 
> https://ckb-explorer-frontend-in-magickbase-repo-okomdwwgi-magickbase.vercel.app/transaction/0x9099747a27e3e4f5064c211a0192323b2268d26044a81369fff51df9e77666f3 image
> 
> "leap_direction": The data for "leapoutBTC" is missing RGB++ tags.

added

**Review by FrederLu — APPROVED** [review]

**Pull request merged**